src/utils/layout/NodesContext.tsx

```diff
@@ -1179,16 +1277,20 @@ export const NodesInternal = {
       [waitForState, node, selector],
     );
   },
-  useNodeDataSelector: () =>
-    Store.useDelayedSelector({
+  useNodeDataSelector: () => {
+    const insideGenerator = GeneratorInternal.useIsInsideGenerator();
+    return Store.useDelayedSelector({
       mode: 'innerSelector',
-      makeArgs: (state) => [((node) => selectNodeData(node, state)) satisfies NodePicker],
-    }),
-  useLaxNodeDataSelector: () =>
-    Store.useLaxDelayedSelector({
+      makeArgs: (state) => [((node) => selectNodeData(node, state, insideGenerator)) satisfies NodePicker],
+    });
+  },
+  useLaxNodeDataSelector: () => {
+    const insideGenerator = GeneratorInternal.useIsInsideGenerator();
+    return Store.useLaxDelayedSelector({
       mode: 'innerSelector',
-      makeArgs: (state) => [((node) => selectNodeData(node, state)) satisfies NodePicker],
-    }),
+      makeArgs: (state) => [((node) => selectNodeData(node, state, insideGenerator)) satisfies NodePicker],
+    });
+  },
```

Comment:
My first thought was that deps was missing, but maybe this does not need deps, as I suppose you cannot "move" in or out of the generator

Reply:
True! :sweat_smile: Also, the `makeArgs` function can't take deps. The delayed selector will re-generate if the hook re-renders.